docs/developers.rst

```diff
@@ -329,3 +329,108 @@ This code marks the test with `com.beam.tags.Periodic` tag. You can also specify
 You can find details about scheduling a continuous integration build under DevOps section `Configure Periodic Jobs`_.
 
 .. _Configure Periodic Jobs: http://beam.readthedocs.io/en/latest/devops.html#configure-periodic-jobs
+
+Scala tips
+^^^^^^^^^^
+Scala Collection
+~~~~~~~~~~~~~~~~
+
+Use ``mutable`` buffer instead of ``immutable var``:
+****************************************************
+
+::
+
+   // Before
+   var buffer = scala.collection.immutable.Vector.empty[Int]
+   buffer = buffer :+ 1
+   buffer = buffer :+ 2
+
+   // After
+   val buffer = scala.collection.mutable.ArrayBuffer.empty[Int]
+   buffer += 1
+   buffer += 2
+
+Don’t create temporary collections, use ``view``:
+*************************************************
+
+::
+
+   val seq: Seq[Int] = Seq(1, 2, 3, 4, 5)
+
+   // Before
+   seq.map(x => x + 2).filter(x => x % 2 == 0).sum
+
+   // After
+   seq.view.map(x => x + 2).filter(x => x % 2 == 0).sum
+
+Don’t emulate ``collectFirst``:
+*******************************
+
+::
+
+   // Get first number >= 4
+   val seq: Seq[Int] = Seq(1, 2, 10, 20)
+   val predicate: Int => Boolean = (x: Int)  => { x >= 4 }
+
+   // Before
+   seq.filter(predicate).headOption
+
+   // After
+   seq.collectFirst { case num if predicate(num) => num }
+
+Don’t emulate ``collect``:
+**************************
+
+::
+
+   // Get first char of string, if it's longer than 3
+   val s: Seq[String] = Seq("C#", "C++", "C", "Scala", "Haskel")
+   val predicate: String => Boolean = (s: String)  => { s.size > 3 }
+
+   // Before
+   s.filter(predicate).map { s => s.head }
+
+   // After
+   s.collect { case curr if predicate(curr) => curr.head }
+
+Prefer not to use ``_1, _2,...`` for ``Tuple`` to improve readability:
```

Comment:
Should this be boiled down to prefer pattern matching? And should there be an example of `val` extraction? 
i.e `val (key, value) = tuple2`

Reply:
Added an example with destructive binding for tuple